Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -4319,7 +4322,30 @@ void InlineCacheCompiler::emitIntrinsicGetter(IntrinsicGetterAccessCase& accessC
 
             ScratchRegisterAllocator::PreservedState preservedState = allocator.preserveReusedRegistersByPushing(jit, ScratchRegisterAllocator::ExtraStackSpace::NoExtraSpace);
 
-            jit.loadTypedArrayByteLength(baseGPR, valueGPR, m_scratchGPR, scratch2GPR, typedArrayType(accessCase.structure()->typeInfo().type()));
+            if (accessCase.intrinsic() == DataViewByteLengthIntrinsic) {
+                auto [outOfBounds, doneCases] = jit.loadDataViewByteLength(baseGPR, valueGPR, m_scratchGPR, scratch2GPR, typedArrayType(accessCase.structure()->typeInfo().type()));
+
+                outOfBounds.link(&jit);
+                InlineCacheCompiler::SpillState spillState = preserveLiveRegistersToStackForCall();
+
+                if (m_stubInfo.useDataIC) {
+                    callSiteIndexForExceptionHandlingOrOriginal();
+                    jit.transfer32(CCallHelpers::Address(m_stubInfo.m_stubInfoGPR, StructureStubInfo::offsetOfCallSiteIndex()), CCallHelpers::tagFor(CallFrameSlot::argumentCountIncludingThis));
+                } else
+                    jit.store32(CCallHelpers::TrustedImm32(callSiteIndexForExceptionHandlingOrOriginal().bits()), CCallHelpers::tagFor(CallFrameSlot::argumentCountIncludingThis));
+
+                jit.makeSpaceOnStackForCCall();
+                jit.storePtr(GPRInfo::callFrameRegister, &m_vm.topCallFrame);
+
+                jit.setupArguments<decltype(operationThrowTypeErrorFrom)>(CCallHelpers::TrustedImmPtr(accessCase.structure()->globalObject()));
+                jit.callOperation<OperationPtrTag>(operationThrowTypeErrorFrom);
+                jit.reclaimSpaceOnStackForCCall();
+
+                restoreLiveRegistersFromStackForCallWithThrownException(spillState);
+                emitExplicitExceptionHandler();
```

Comment:
Let's make IC code more simple and tight. I think you can just jump to `m_failAndIgnore` at the end.
You also need to do `allocator.restoreReusedRegistersByPopping` before that jump. And these code generation should be done after `succeed()` call to make hot path code tight.
See `AccessCase::IndexedInt32Load`'s `m_failAndIgnore` jump code for example.